infinite-tracing/src/main/java/com/newrelic/ResponseObserver.java

```diff
@@ -15,18 +15,21 @@
     private final Logger logger;
     private final DisconnectionHandler disconnectionHandler;
     private final AtomicBoolean shouldRecreateCall;
+    private final AtomicBoolean firstFailedPrecondition;
 
     public ResponseObserver(MetricAggregator metricAggregator, Logger logger, DisconnectionHandler disconnectionHandler,
-            AtomicBoolean shouldRecreateCall) {
+                            AtomicBoolean shouldRecreateCall, AtomicBoolean firstFailedPrecondition) {
         this.metricAggregator = metricAggregator;
         this.logger = logger;
         this.disconnectionHandler = disconnectionHandler;
         this.shouldRecreateCall = shouldRecreateCall;
+        this.firstFailedPrecondition = firstFailedPrecondition;
     }
 
     @Override
     public void onNext(V1.RecordStatus value) {
         metricAggregator.incrementCounter("Supportability/InfiniteTracing/Response");
+        firstFailedPrecondition.compareAndSet(false,true);
```

Comment:
According to the agent spec, receiving gRPC `Ok` codes are an acceptable indicator to reset  failed precondition  so the next time the error occurs, it is considered the first one again.